runtime/sema/type_test.go

```diff
@@ -529,3 +529,71 @@ func TestBeforeType_Strings(t *testing.T) {
 		beforeType.QualifiedString(),
 	)
 }
+
+func TestQualifiedIdentifierCreation(t *testing.T) {
```

Comment:
Thank you for adding a unit-test 👍 

Could you please add another case where the container type is nil? (for coverage)